bpf/process/types/tuple.h

```diff
@@ -7,13 +7,39 @@
 #define AF_INET	 2
 #define AF_INET6 10
 
+#define IPV4LEN 4
+#define IPV6LEN 16
+
 struct tuple_type {
-	__u32 saddr;
-	__u32 daddr;
+	__u64 saddr[2];
+	__u64 daddr[2];
 	__u16 sport;
 	__u16 dport;
 	__u16 protocol;
 	__u16 family;
 };
 
+static inline __attribute__((always_inline)) void
+write_ipv6_addr_from_ipv4(u64 *dest, u32 src)
+{
+	dest[0] = src;
+	dest[1] = 0;
```

Comment:
AFAICT, we keep everything in host order, but I'm not sure if above works for big-endian hosts.
I guess it does not matter much in practice before even for ARM. I think it supports both modes, and most people will run it as little endian.

Reply:
We store everything in network order, but we don't have any code to support big endian currently. I don't think it's a big deal, but it would be relatively easy to add if it became important.

Reply:
Do we? I guess I misread the code. Let me check...

I guess that's what I saw:
>	event->tuple.sport = bpf_ntohs(event->tuple.sport);
>	event->tuple.dport = bpf_ntohs(event->tuple.dport);

So eveht->tuple has addresses in network order and ports in host order? Or am I missing something again?

Reply:
No you're correct, I was only thinking about IP addresses. Ports are indeed in host order.